// @pulumi/gcp/storage/notification.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates a new notification configuration on a specified bucket, establishing a flow of event notifications from GCS to a Cloud Pub/Sub topic.
 *  For more information see
 * [the official documentation](https://cloud.google.com/storage/docs/pubsub-notifications)
 * and
 * [API](https://cloud.google.com/storage/docs/json_api/v1/notifications).
 *
 * In order to enable notifications, a special Google Cloud Storage service account unique to the project
 * must exist and have the IAM permission "projects.topics.publish" for a Cloud Pub/Sub topic in the project.
 * This service account is not created automatically when a project is created.
 * To ensure the service account exists and obtain its email address for use in granting the correct IAM permission, use the
 * [`gcp.storage.getProjectServiceAccount`](https://www.terraform.io/docs/providers/google/d/storage_project_service_account.html)
 * datasource's `emailAddress` value, and see below for an example of enabling notifications by granting the correct IAM permission.
 * See [the notifications documentation](https://cloud.google.com/storage/docs/gsutil/commands/notification) for more details.
 *
 * > **NOTE**: This resource can affect your storage IAM policy. If you are using this in the same config as your storage IAM policy resources, consider
 * making this resource dependent on those IAM resources via `dependsOn`. This will safeguard against errors due to IAM race conditions.
 */
class Notification extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["bucket"] = state ? state.bucket : undefined;
            inputs["customAttributes"] = state ? state.customAttributes : undefined;
            inputs["eventTypes"] = state ? state.eventTypes : undefined;
            inputs["notificationId"] = state ? state.notificationId : undefined;
            inputs["objectNamePrefix"] = state ? state.objectNamePrefix : undefined;
            inputs["payloadFormat"] = state ? state.payloadFormat : undefined;
            inputs["selfLink"] = state ? state.selfLink : undefined;
            inputs["topic"] = state ? state.topic : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.bucket === undefined) {
                throw new Error("Missing required property 'bucket'");
            }
            if (!args || args.payloadFormat === undefined) {
                throw new Error("Missing required property 'payloadFormat'");
            }
            if (!args || args.topic === undefined) {
                throw new Error("Missing required property 'topic'");
            }
            inputs["bucket"] = args ? args.bucket : undefined;
            inputs["customAttributes"] = args ? args.customAttributes : undefined;
            inputs["eventTypes"] = args ? args.eventTypes : undefined;
            inputs["objectNamePrefix"] = args ? args.objectNamePrefix : undefined;
            inputs["payloadFormat"] = args ? args.payloadFormat : undefined;
            inputs["topic"] = args ? args.topic : undefined;
            inputs["notificationId"] = undefined /*out*/;
            inputs["selfLink"] = undefined /*out*/;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(Notification.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Notification resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Notification(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Notification.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Notification.__pulumiType;
    }
}
exports.Notification = Notification;
/** @internal */
Notification.__pulumiType = 'gcp:storage/notification:Notification';
