// @pulumi/gcp/iap/brand.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * OAuth brand data. Only "Organization Internal" brands can be created
 * programatically via API. To convert it into an external brands
 * please use the GCP Console.
 *
 * > **Note:** Brands can be created only once for a Google Cloud Platform
 * project and cannot be deleted. Destroying a provider-managed Brand
 * will remove it from state but *will not delete the resource on the server.*
 *
 * ## Example Usage
 */
class Brand extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["applicationTitle"] = state ? state.applicationTitle : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["orgInternalOnly"] = state ? state.orgInternalOnly : undefined;
            inputs["project"] = state ? state.project : undefined;
            inputs["supportEmail"] = state ? state.supportEmail : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.applicationTitle === undefined) {
                throw new Error("Missing required property 'applicationTitle'");
            }
            if (!args || args.supportEmail === undefined) {
                throw new Error("Missing required property 'supportEmail'");
            }
            inputs["applicationTitle"] = args ? args.applicationTitle : undefined;
            inputs["project"] = args ? args.project : undefined;
            inputs["supportEmail"] = args ? args.supportEmail : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["orgInternalOnly"] = undefined /*out*/;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(Brand.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Brand resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Brand(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Brand.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Brand.__pulumiType;
    }
}
exports.Brand = Brand;
/** @internal */
Brand.__pulumiType = 'gcp:iap/brand:Brand';
