// @pulumi/gcp/bigquery/job.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Jobs are actions that BigQuery runs on your behalf to load data, export data, query data, or copy data.
 * Once a BigQuery job is created, it cannot be changed or deleted.
 *
 * ## Example Usage
 */
class Job extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["copy"] = state ? state.copy : undefined;
            inputs["extract"] = state ? state.extract : undefined;
            inputs["jobId"] = state ? state.jobId : undefined;
            inputs["jobTimeoutMs"] = state ? state.jobTimeoutMs : undefined;
            inputs["jobType"] = state ? state.jobType : undefined;
            inputs["labels"] = state ? state.labels : undefined;
            inputs["load"] = state ? state.load : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["project"] = state ? state.project : undefined;
            inputs["query"] = state ? state.query : undefined;
            inputs["userEmail"] = state ? state.userEmail : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.jobId === undefined) {
                throw new Error("Missing required property 'jobId'");
            }
            inputs["copy"] = args ? args.copy : undefined;
            inputs["extract"] = args ? args.extract : undefined;
            inputs["jobId"] = args ? args.jobId : undefined;
            inputs["jobTimeoutMs"] = args ? args.jobTimeoutMs : undefined;
            inputs["labels"] = args ? args.labels : undefined;
            inputs["load"] = args ? args.load : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["project"] = args ? args.project : undefined;
            inputs["query"] = args ? args.query : undefined;
            inputs["jobType"] = undefined /*out*/;
            inputs["userEmail"] = undefined /*out*/;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(Job.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Job resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Job(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Job.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Job.__pulumiType;
    }
}
exports.Job = Job;
/** @internal */
Job.__pulumiType = 'gcp:bigquery/job:Job';
