// @pulumi/gcp/appengine/domainMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A domain serving an App Engine application.
 *
 * To get more information about DomainMapping, see:
 *
 * * [API documentation](https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.domainMappings)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/appengine/docs/standard/python/mapping-custom-domains)
 *
 * ## Example Usage
 */
class DomainMapping extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["domainName"] = state ? state.domainName : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["overrideStrategy"] = state ? state.overrideStrategy : undefined;
            inputs["project"] = state ? state.project : undefined;
            inputs["resourceRecords"] = state ? state.resourceRecords : undefined;
            inputs["sslSettings"] = state ? state.sslSettings : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.domainName === undefined) {
                throw new Error("Missing required property 'domainName'");
            }
            inputs["domainName"] = args ? args.domainName : undefined;
            inputs["overrideStrategy"] = args ? args.overrideStrategy : undefined;
            inputs["project"] = args ? args.project : undefined;
            inputs["sslSettings"] = args ? args.sslSettings : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["resourceRecords"] = undefined /*out*/;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(DomainMapping.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DomainMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DomainMapping(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DomainMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainMapping.__pulumiType;
    }
}
exports.DomainMapping = DomainMapping;
/** @internal */
DomainMapping.__pulumiType = 'gcp:appengine/domainMapping:DomainMapping';
