// @pulumi/gcp/iot/device.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Google Cloud IoT Core device.
 *
 * To get more information about Device, see:
 *
 * * [API documentation](https://cloud.google.com/iot/docs/reference/cloudiot/rest/)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/iot/docs/)
 *
 * ## Example Usage
 */
class Device extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["blocked"] = state ? state.blocked : undefined;
            inputs["config"] = state ? state.config : undefined;
            inputs["credentials"] = state ? state.credentials : undefined;
            inputs["gatewayConfig"] = state ? state.gatewayConfig : undefined;
            inputs["lastConfigAckTime"] = state ? state.lastConfigAckTime : undefined;
            inputs["lastConfigSendTime"] = state ? state.lastConfigSendTime : undefined;
            inputs["lastErrorStatus"] = state ? state.lastErrorStatus : undefined;
            inputs["lastErrorTime"] = state ? state.lastErrorTime : undefined;
            inputs["lastEventTime"] = state ? state.lastEventTime : undefined;
            inputs["lastHeartbeatTime"] = state ? state.lastHeartbeatTime : undefined;
            inputs["lastStateTime"] = state ? state.lastStateTime : undefined;
            inputs["logLevel"] = state ? state.logLevel : undefined;
            inputs["metadata"] = state ? state.metadata : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["numId"] = state ? state.numId : undefined;
            inputs["registry"] = state ? state.registry : undefined;
            inputs["state"] = state ? state.state : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.registry === undefined) {
                throw new Error("Missing required property 'registry'");
            }
            inputs["blocked"] = args ? args.blocked : undefined;
            inputs["credentials"] = args ? args.credentials : undefined;
            inputs["gatewayConfig"] = args ? args.gatewayConfig : undefined;
            inputs["logLevel"] = args ? args.logLevel : undefined;
            inputs["metadata"] = args ? args.metadata : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["registry"] = args ? args.registry : undefined;
            inputs["config"] = undefined /*out*/;
            inputs["lastConfigAckTime"] = undefined /*out*/;
            inputs["lastConfigSendTime"] = undefined /*out*/;
            inputs["lastErrorStatus"] = undefined /*out*/;
            inputs["lastErrorTime"] = undefined /*out*/;
            inputs["lastEventTime"] = undefined /*out*/;
            inputs["lastHeartbeatTime"] = undefined /*out*/;
            inputs["lastStateTime"] = undefined /*out*/;
            inputs["numId"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(Device.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Device resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Device(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Device.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Device.__pulumiType;
    }
}
exports.Device = Device;
/** @internal */
Device.__pulumiType = 'gcp:iot/device:Device';
