// @pulumi/gcp/identityplatform/tenantInboundSamlConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Inbound SAML configuration for a Identity Toolkit tenant.
 *
 * You must enable the
 * [Google Identity Platform](https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity) in
 * the marketplace prior to using this resource.
 *
 * ## Example Usage
 */
class TenantInboundSamlConfig extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        if (opts && opts.id) {
            const state = argsOrState;
            inputs["displayName"] = state ? state.displayName : undefined;
            inputs["enabled"] = state ? state.enabled : undefined;
            inputs["idpConfig"] = state ? state.idpConfig : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["project"] = state ? state.project : undefined;
            inputs["spConfig"] = state ? state.spConfig : undefined;
            inputs["tenant"] = state ? state.tenant : undefined;
        }
        else {
            const args = argsOrState;
            if (!args || args.displayName === undefined) {
                throw new Error("Missing required property 'displayName'");
            }
            if (!args || args.idpConfig === undefined) {
                throw new Error("Missing required property 'idpConfig'");
            }
            if (!args || args.spConfig === undefined) {
                throw new Error("Missing required property 'spConfig'");
            }
            if (!args || args.tenant === undefined) {
                throw new Error("Missing required property 'tenant'");
            }
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["idpConfig"] = args ? args.idpConfig : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["project"] = args ? args.project : undefined;
            inputs["spConfig"] = args ? args.spConfig : undefined;
            inputs["tenant"] = args ? args.tenant : undefined;
        }
        if (!opts) {
            opts = {};
        }
        if (!opts.version) {
            opts.version = utilities.getVersion();
        }
        super(TenantInboundSamlConfig.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TenantInboundSamlConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TenantInboundSamlConfig(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TenantInboundSamlConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TenantInboundSamlConfig.__pulumiType;
    }
}
exports.TenantInboundSamlConfig = TenantInboundSamlConfig;
/** @internal */
TenantInboundSamlConfig.__pulumiType = 'gcp:identityplatform/tenantInboundSamlConfig:TenantInboundSamlConfig';
